deprish_to_eia.py

```diff
@@ -0,0 +1,287 @@
+        'technology_description', 'ferc_acct_name', 'utility_id_eia',
+        'true_gran', 'appro_part_label', 'appro_record_id_eia',
+    ]
+    logger.info("Merging fuzzy matches.")
+    match_merge = pd.merge(
+        get_fuzzy_matches(df1, df2, key1, key2, threshold=75)[cols1],
+        df2.reset_index().drop_duplicates(
+            subset=['report_year', 'plant_name_new'])[cols2],
+        left_on=['report_year', 'plant_name_match'],
+        right_on=['report_year', key2], how='left')
+    logger.info(f"Mathing resulted in {len(match_merge)} connections.")
+    return match_merge
+
+
+def match_deprish_eia(file_path_mul, file_path_deprish, sheet_name_deprish):
+    """Match."""
+    plant_parts_df = plant_part_agg_eia.get_master_unit_list_eia(file_path_mul)
+
+    key1 = 'plant_name'
+    key2 = 'plant_name_new'
+    df1 = prep_deprish(file_path_deprish, sheet_name_deprish,
+                       plant_parts_df, key1)
+    df2 = prep_master_parts_eia(plant_parts_df, key2)
+    return match_merge(df1, df2, key1, key2)
+
+###############################################################################
+# EXPORT
+###############################################################################
+
+
+def generate_depreciation_matches(file_path_mul,
+                                  file_path_deprish,
+                                  sheet_name_deprish,
+                                  sheet_name_output):
+    """
+    Generate the matched names and save to execl.
+
+    Args:
+         file_path_mul (path-like): path to the master unit list/.
+         file_path_deprish (path-like): path to the excel workbook which
+            contains depreciation data.
+         sheet_name_deprish (string): name of the excel tab which contains the
+            plant names from the depreciation data.
+         sheet_name_output (string): name of the excel tab which the matches
+            will be output.
+    """
+    if not file_path_deprish.is_file():
+        raise AssertionError(
+            f"File does not exist: {file_path_deprish}"
+            "Depretiation file must exist."
+        )
+
+    deprish_match = match_deprish_eia(
+        file_path_mul, file_path_deprish,
+        sheet_name_deprish=sheet_name_deprish)
+
+    save_to_workbook(deprish_match, file_path=file_path_deprish,
+                     sheet_name=sheet_name_output)
+    return deprish_match
+
+
+def save_to_workbook(df, file_path, sheet_name):
+    """
+    Save a dataframe to an existing workbook.
+
+    Args:
+        df (pandas.DataFrame): the table you want to export.
+        file_path (path-like): the location of the excel workbook.
+        sheet_name (string): the name of your new sheet. If this sheet exists
+            in your workbook, the new sheet will be sheet_name_1.
+    """
+    logger.info(f"Saving dataframe to {file_path}")
+    if not file_path.exists():
+        raise AssertionError(f'file path {str(file_path)} does not exist')
```

Comment:
You shouldn't have to cast `file_path` to type `str` here -- the f-string formatter automatically does this for anything that's being embedded in the string.